lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -1291,6 +1320,33 @@ def test_certificate_custom_language_template_with_locale_language_from_catalogu
         self.assertEqual(response.status_code, 200)
         self.assertContains(response, 'course name: test_right_lang_template')
 
+    @override_settings(FEATURES=FEATURES_WITH_CUSTOM_CERTS_ENABLED)
+    @patch('certificates.views.webview.get_course_run_details')
+    @patch('certificates.api.get_course_organization_id')
+    def test_certificate_custom_template_with_hours_of_effort(self, mock_get_org_id, mock_get_course_run_details):
+        """
+        Tests custom template renders properly with Hours of Effort data
+        """
+        mock_get_course_run_details.return_value = self.mock_course_run_details
+        CertificateGenerationCourseSetting.set_include_hours_of_effort_for_course(self.course.id, True)
+        CertificateGenerationCourseSetting.set_language_specific_templates_enabled_for_course(self.course.id, True)
+        self._add_course_certificates(count=1, signatory_count=2)
+        self._create_custom_template_with_hours_of_effort(org_id=1, language=None)
+        with patch.dict("django.conf.settings.FEATURES", {
+            "CERTIFICATES_HTML_VIEW": True,
+        }):
+            test_url = get_certificate_url(
+                user_id=self.user.id,
+                course_id=unicode(self.course.id)
+            )
+        with patch('django.http.HttpRequest.build_absolute_uri') as mock_abs_uri:
+            mock_abs_uri.return_value = '='.join(['http://localhost/?param', u'é'])
```

Comment:
What is this URL? Also since there's only one param, I don't think there's a need for `join`.